bluemira/radiation_transport/flux_surfaces_maker.py

```diff
@@ -0,0 +1,303 @@
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2021-2023 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh,
+#                         J. Morris, D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+"""
+A simplified 2-D solver for calculating charged particle heat loads.
+"""
+
+from copy import deepcopy
+from dataclasses import dataclass, fields
+from typing import Dict
+
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.axes import Axes
+
+from bluemira.base.constants import EPS
+from bluemira.base.look_and_feel import bluemira_warn
+from bluemira.display.plotter import plot_coordinates
+from bluemira.equilibria.find import find_flux_surface_through_point
+from bluemira.equilibria.flux_surfaces import OpenFluxSurface
+from bluemira.geometry.coordinates import Coordinates, coords_plane_intersect
+from bluemira.geometry.plane import BluemiraPlane
+from bluemira.radiation_transport.error import AdvectionTransportError
+
+__all__ = ["FluxSurfaceMaker"]
+
+
+class FluxSurfaceMaker:
+    """
+    A simplified charged particle transport model along open field lines.
+
+    Parameters
+    ----------
+    config: Dict[str, float]
+        The parameters for running the transport model. See
+        :class:`ChargedParticleSolverParams` for available parameters
+        and their defaults.
+    equilibrium: Equilibrium
+        The equilibrium defining flux surfaces.
+    dx_mp: float (optional)
+        The midplane spatial resolution between flux surfaces [m]
+        (default: 0.001).
+    """
+
+    def __init__(self, equilibrium, dx_mp: float = 0.001):
+        self.eq = equilibrium
+        self.dx_mp = dx_mp
+
+        # Constructors
+        self.first_wall = None
+        self.flux_surfaces_ob_lfs = None
+        self.flux_surfaces_ob_hfs = None
+        self.flux_surfaces_ib_lfs = None
+        self.flux_surfaces_ib_hfs = None
+        self.x_sep_omp = None
+        self.x_sep_imp = None
```

Comment:
I think all of these should be calculated on init based on whether the provided equilibrium is an single null or a double null.

If the equilibrium is changed they can be recalculated but otherwise the first wall changing is the only thing analysed therefore everything else can be setup beforehand.

Reply:
A few line became a big change 😅 

Reply:
yeah sorry 😇